### DISCUSSION TITLE
how to customize the showPasswordIcon() and hidePasswordIcon() color?

### DISCUSSION THREAD

**Discussion by devhoussam1998** [rawilk/filament-password-input#2] [Q&A]
there is no 

```
->showPasswordIconColor('success'),
->hidePasswordIconColor('success'),
```

**Comment by rawilk** [comment]
There is a `fi-pw-reveal-icon` css class on the password visibility toggle element that you can override in your css files. I may consider making it possible to change the icon color via php in a future release though.

**Reply by devhoussam1998** [reply]
still can't use 

```
->showPasswordIconColor('success'),
->hidePasswordIconColor('success'),
```

**Reply by rawilk** [reply]
@devhoussam1998 Feel free to PR this if it's important to you. There are periods of time where I'm not able to dedicate a lot of time to my open source work, so features that aren't critical for myself are going to be pushed further down the priority list.

**Reply by rawilk** [reply]
I've just tagged v2.0.0, which makes this possible, although in a slightly different way. Since v2 is using filament's new revealable feature on inputs, you can customize the `ShowPasswordAction` and `HidePasswordAction` actions in a service provider to change the color of the icons, and more.

```php
use Filament\Forms\Components\TextInput\Actions\ShowPasswordAction;

ShowPasswordAction::configureUsing(function (ShowPasswordAction $action) {
    $action->color('primary');
}, isImportant: true);
```

Note, be sure to set the `isImportant` flag to true, otherwise this won't work (at least at the time of this writing).
